### PR TITLE
Update libnfc and libmifare, add option to disable building examples

### PR DESCRIPTION
**Pull request by Mstrodl**
Modified from Nick's branch

### CODE REVIEW

**Review by nicholastmosher — APPROVED** [review]
Looks good to me! For anybody else wanting to try, install libnfc and probably libfreefare and then run

```
mkdir build && cd build && cmake ..
```

My output comes out like this, which looks good

```
-- The C compiler identification is GNU 9.3.1
-- Check for working C compiler: /bin/cc
-- Check for working C compiler: /bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /bin/pkg-config (found version "1.6.3")
-- Found LibNfc: /usr/lib64/libnfc.so
-- Found LibFreefare: /usr/local/lib/libfreefare.so
-- Found UUID: /usr/lib64/libuuid.so
-- Found OpenSSL: /usr/lib64/libcrypto.so (found version "1.1.1g")
-- Configuring done
-- Generating done
-- Build files have been written to: /home/nick/Documents/libgatekeeper/build
```

File: examples/format-tag.c [hunk 1]
```diff
@@ -52,7 +52,7 @@ int main(int argc, const char *argv[]) {
         MifareTag tag = tags[i];
 
         switch (freefare_get_tag_type(tag)) {
-            case DESFIRE:
+            case MIFARE_DESFIRE:
```
Comment (stevenmirabito) [inline examples/format-tag.c:55]: Did you build `libfreefare` from master or from the latest release (0.4.0)? The symbols changed in master as part of a larger refactor but afaik that's still a WIP and things will not work if you try to run against it.

Comment (Mstrodl) [reply examples/format-tag.c:55]: Oh really? I did update it because libfreefare git has support for cmake...

**Comment by Mstrodl** [comment]
Newer mifare/libnfc break stuff still

**Pull request closed without merging**